js/userscript.js

```diff
@@ -1,51 +1,253 @@
+    signUpForm.style.display = 'block';
+  });
 
-Array.prototype.slice.call(forms)
-    .forEach(function (form) {
-    form.addEventListener('submit', function (event) {
-        if (!form.checkValidity()){
-        event.preventDefault()
-        event.stopPropagation()
-        } else if (form.checkValidity()) {
-          saveUser()  
-        }
-        // event.preventDefault()
-        form.classList.add('was-validated')
-        
-    } )
-}, false);
+  signInFormTab.addEventListener('click', (e) => {
+    signUpForm.style.display = 'none';
+    // signInForm.style.display = 'block';
+  });
+  
+  signUpForm.addEventListener('submit', (event) => {
+    if (!signUpForm.checkValidity()) {
+      event.preventDefault();
+      event.stopPropagation();
+    } else if (signUpForm.checkValidity()) {
+      console.log(signUpForm);
+      saveUser();
+    }
+    signUpForm.classList.add('was-validated');
+  }), false;
 
+  signInForm.addEventListener('submit', (event) => {
+    event.preventDefault();
+    checkUser();
+  });
+}
 
+// SAVE USER CREATED
 function saveUser() {
-  let userEmailInput = document.getElementById('userEmailInput').value;
-  let userEmailInputStyle = document.getElementById('userEmailInput').classList;
-  let userEmailPassword = document.getElementById('userPasswordInput').value;
+  let signInForm = document.getElementById('signInForm');
+  // signUpForm.style.display = 'block';
+
+  let users = [];
+  if (localStorage.getItem('users')) {
+    let localStorageUsers = JSON.parse(localStorage.getItem('users'));
+    users = localStorageUsers.map((user) => user);
+    console.log(users);
+  }
+  let userNameInput = document.getElementById('signUpUserName').value;
+  let userEmailInput = document.getElementById('signUpUserEmailInput').value;
+  let userEmailInputStyle = document.getElementById('signUpUserEmailInput').classList;
+  let userEmailPassword = document.getElementById('signUpUserPasswordInput').value;
 
-  console.log(userEmailInputStyle);
   const user = {
-    userName: userEmailInput,
+    userName: userNameInput,
+    userEmail: userEmailInput,
     userPassword: userEmailPassword,
-    signedIn: true,
+    signedIn: false,
+    favShows: [],
+    episodeTracker: [],
   };
-  localStorage.setItem("user", JSON.stringify(user));
-  userPage();
+
+  users.push(user);
+  console.log(users);
+  localStorage.setItem("users", JSON.stringify(users));
+  
+  // userPage();
+}
+
+function checkUser() {
+  let signInInvalid = document.getElementById('signInInvalid');
+  let userNameEmailInput = document.getElementById('signInUserNameEmail').value;
+  let userPassword = document.getElementById('signInUserPasswordInput').value;
+  let userPasswordInput = document.getElementById('signInUserPasswordInput');
+
+  let signedInUserTrue = [];
+  if (localStorage.getItem('users')) {
+    let localStorageUsers = JSON.parse(localStorage.getItem('users'));
+    users = localStorageUsers.map((user) => {
```

Comment:
You're setting `users` as a global variable here, not sure if intentional but it's best to keep variables local where possible (using `let` or `const`).